sphinx_gallery/directives.py

```diff
@@ -2,11 +2,14 @@
 Custom Sphinx directives
 ========================
 """
-
 import os
+from pathlib import Path
```

Comment:
Looking at your test, the failure is that some `path/to/file.ext` fails. It's almost certainly related to this -- somewhere you need to ensure that the URI that ends up in the `img src="..."` HTML tag is constructed using the Posix convention rather than the system native convention (which will be wrong on Windows because it will use `\` instead of `/`). I'd expect you to need to use `PosixPath` instead of `Path` somewhere to fix it.